src/Parser/Template.cs

```diff
@@ -54,6 +57,95 @@ public Template(string template, bool isPositional, List<Literal> literals, List
       Holes = holes.ToArray();
     }
 
+    public string Render(object[] parameters)
+    {
+        var sb = new StringBuilder();
```

Comment:
> Could we use Value.Length + constant * Holes.Length for some arbitrary constant?

That's also a nice approach. I was thinking it's better a bit to large then too small. Also, if it's to small then the stringbuilder would copy it in an array twice the size (the *2)

Also, if you